share/p2p/peers/pool.go

```diff
@@ -0,0 +1,153 @@
+package peers
+
+import (
+	"context"
+	"sync"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+)
+
+// pool stores peers and provides methods for simple round-robin access.
+type pool struct {
+	m           sync.Mutex
+	peersList   []peer.ID
+	active      map[peer.ID]bool
+	activeCount int
+	next        int
+
+	hasPeer   bool
+	hasPeerCh chan struct{}
+
+	cleanupDisabled bool
```

Comment:
let's remove this, it should always be enabled

Reply:
Need it for tests